src/NUnitFramework/tests/Constraints/CollectionEquivalentConstraintTests.cs

```diff
@@ -152,8 +152,19 @@ public void EquivalentHonorsUsing()
                 .Using<string>((x, y) => StringUtil.Compare(x, y, true))
                 .ApplyTo(set2).IsSuccess);
         }
+#if (NET_4_0 || NET_4_5) && !PORTABLE
 
-#if ( NET_4_0 || NET_4_5 ) && !PORTABLE
+        [Test]
+        public void EquivalentHonorsUsingWhenCollectionsAreOfDifferentTypes()
+        {
+            ICollection set1 = new SimpleObjectCollection("1", "2", "3");
+            ICollection set2 = new SimpleObjectCollection(1, 2, 3);
+
+            Assert.That(set2, Is.EquivalentTo(set1).Using<int, string>((i, s) => i.ToString() == s));
+        }
+#endif
```

Comment:
Well, you have already implemented it in all those classes. We just don't have tests. :-)

I agree that the particular int to string comparison would not be so useful to UniqueItemsConstraint, but being able to override how equality comparison is done is useful and it already supports all the other overloads of Using. The thing is, I don't want users worrying about which overloads work for which constraints - they should all work.